src/Filter/Validator.php

```diff
@@ -32,7 +32,7 @@ class Validator implements FilterInterface
      */
     public function __construct($validator)
     {
-        if ($validator instanceof Traversable) {
+        if ($validator instanceof Traversable && ! $validator instanceof LaminasValidator) {
```

Comment:
@gsteel this is an interesting case study of a scenario where **adding** an interface implementation to an existing system provokes a BC break :D